Add support for adm1281

jsanbuen opened the pull request:
## PR Description

- Please replace this comment with a summary of your changes, and add any context
necessary to understand them. List any dependencies required for this change.
- To check the checkboxes below, insert a 'x' between square brackets (without
any space), or simply check them after publishing the PR.
- If you changes include a breaking change, please specify dependent PRs in the
description and try to push all related PRs simultaneously.

## PR Type
- [ ] Bug fix (a change that fixes an issue)
- [X] New feature (a change that adds new functionality)
- [ ] Breaking change (a change that affects other repos or cause CIs to fail)

## PR Checklist
- [X] I have conducted a self-review of my own code changes
- [X] I have tested the changes on the relevant hardware
- [ ] I have updated the documentation outside this repo accordingly (if there is the case)

jsanbuen commented:
changelog
-addressed compiler warnings (if-else bracketing and switch case fall through)
-checkpatch error is from user name mismatch, due to difference in user.name config in time of initial commit (user.name is jsanbuen) vs. during rebase of commits for revision (user.name changed to Jose Ramon San Buenaventura)

jsanbuen commented:
changelog v3:
-modified commit settings to match intended config for user.name (originally using jsanbuen but changed to Jose Ramon San Buenaventura)

nunojsa commented on Documentation/devicetree/bindings/hwmon/adi,adm1275.yaml:
```diff
@@ -94,6 +95,7 @@ allOf:
           contains:
             enum:
               - adi,adm1278
+              - adi,adm1281
```
as for your git subject maybe just state: 

`Add support for the adm1281 Hot-Swap Controller and Digital Power and Energy Monitor `

nunojsa reviewed (commented):
LGTM,

As of the bindings patch I would likely rephrase a bit your commit message. Don say "identifier". Just focus that you're supporting adm1281 which is a similar hot swap controller as the adm1275 driver. Then mention that it additionally support STATUS_CML reads so that is also being added.

Note that I'm not approving the PR just because I want (would like) to see this first upstreamed. So, please go ahead and send the patches :)

nunojsa commented:
@jsanbuen , I see this was already applied upstream. Can you sync this PULL with the accepted pacthes (just cherry-pick them from [here](https://git.kernel.org/pub/scm/linux/kernel/git/groeck/linux-staging.git/log/?h=hwmon-next)) so we can merge this?

jsanbuen commented:
Note: patch sent upstream now applied based on email from maintainer
Note: patch sent upstream was based on torvalds repository version of adm1275.c

changelog:
-removed extra boolean member in adm1275_data structure (have_status_cml)
-removed case for STATUS_CML in adm1275_read_byte_data
-refactored commit message
